src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/MethodLdtokenNode.cs

```diff
@@ -0,0 +1,181 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.Collections.Generic;
+
+using ILCompiler.DependencyAnalysisFramework;
+using Internal.Text;
+using Internal.TypeSystem;
+
+namespace ILCompiler.DependencyAnalysis
+{
+    /// <summary>
+    /// This method node is used for dependency tracking, like in the case of GVMs for example.
+    /// </summary>
+    internal class MethodLdtokenNode : DependencyNodeCore<NodeFactory>
+    {
+        private MethodDesc _method;
+
+        public MethodLdtokenNode(MethodDesc method)
+        {
+            Debug.Assert(!method.IsRuntimeDeterminedExactMethod);
+            _method = method;
+        }
+
+        public override bool HasConditionalStaticDependencies => false;
+        public override bool InterestingForDynamicDependencyAnalysis => false;
+        public override bool StaticDependenciesAreComputed => true;
+        protected override string GetName() => "__MethodLdtokenNode_" + NodeFactory.NameMangler.GetMangledMethodName(_method);
+
+        public override IEnumerable<DependencyListEntry> GetStaticDependencies(NodeFactory context)
+        {
+            return Array.Empty<DependencyListEntry>();
+        }
+        public override IEnumerable<CombinedDependencyListEntry> GetConditionalStaticDependencies(NodeFactory context)
+        {
+            return Array.Empty<CombinedDependencyListEntry>();
+        }
+
+        public override bool HasDynamicDependencies
+        {
+            get
+            {
+                // Only generic virtual methods work have dynamic dependencies
+                if (_method.HasInstantiation && _method.IsVirtual)
```

Comment:
This node is just used for the GVM analysis for now, so it would make sense for now to have this as an assert in the constructor. I don't fully know yet if this node will be used for other kinds of dependency analysis. This is related to my question about renaming this node to something different that reflects it's only for GVMs, and I would vote that if we need to add another kind of dependency analysis, put it in a different new node.
@davidwrighton, do you agree?

---
In reply to: [96539338](https://github.com/dotnet/corert/pull/2521#discussion_r96539338) [](ancestors = 96539338)